src/main/java/mate/academy/springboot/datajpa/repository/specification/ProductSpecificationManager.java

```diff
@@ -0,0 +1,28 @@
+package mate.academy.springboot.datajpa.repository.specification;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductSpecificationManager implements SpecificationManager<Product> {
+    private final Map<String, SpecificationProvider<Product>> providerMap;
+
+    @Autowired
+    public ProductSpecificationManager(List<SpecificationProvider<Product>> providerList) {
+        this.providerMap = providerList.stream()
+                .collect(Collectors.toMap(SpecificationProvider::getFilterKey, n -> n));

```

Comment:
let's use Function.identity, as a cool java guys

Reply:
🤦 remember that black guy :"we are not the same"?